docker.go

```diff
@@ -281,6 +363,64 @@ func (p Plugin) Exec() error {
 	return nil
 }
 
+func parseCacheMetrics(ch <-chan string) (CacheMetrics, error) {
+	var cacheMetrics CacheMetrics
+	cacheMetrics.Layers = make(map[int]LayerStatus) // Initialize the map
+
+	re := regexp.MustCompile(`#(\d+) (DONE|CACHED|ERRORED|CANCELLED)(?: ([0-9.]+)s)?`)
+
+	for line := range ch {
+		matches := re.FindAllStringSubmatch(line, -1)
+		for _, match := range matches {
+			if len(match) > 2 {
```

Comment:
Add an early continue to save on indentation

Reply:
Done